File: android_java_hummus/src/main/java/com/github/klee0kai/hummus/adapterdelegates/delegate/viewbinding/ViewBindingHolder.java
package com.github.klee0kai.hummus.adapterdelegates.delegate.viewbinding;

import androidx.recyclerview.widget.RecyclerView;
import androidx.viewbinding.ViewBinding;

public class ViewBindingHolder<T extends ViewBinding> extends RecyclerView.ViewHolder {

    public final T binding;

    public ViewBindingHolder(T binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

}

File: android_kotlin_hummus/src/main/kotlin/com/github/klee0kai/hummus/adapterdelegates/delegate/viewbinding/ViewBindingHolder.kt
package com.github.klee0kai.hummus.adapterdelegates.delegate.viewbinding

import androidx.recyclerview.widget.RecyclerView.ViewHolder
import androidx.viewbinding.ViewBinding

open class ViewBindingHolder<T : ViewBinding>(
    open val binding: T
) : ViewHolder(binding.root)

